examples/android_app/README.md

```diff
@@ -0,0 +1,146 @@
+* First, assuming you have the model and tokenizer ready in ExecuTorch format. You can get more detail [here](https://github.com/pytorch/executorch/blob/main/examples/demo-apps/android/LlamaDemo/docs/delegates/xnnpack_README.md#prepare-models) on how to prepare them. Once you have the model and tokenizer ready, you can push them to the device by:
+```
+adb shell mkdir -p /data/local/tmp/llama
+adb push llama.pte /data/local/tmp/llama
+adb push tokenizer.bin /data/local/tmp/llama
+```
+* Then, select the appropriate Backend(XNNPACK on CPUs), Model, Tokenizer and Model Type
+* Click `Load Model` and this will take you back to the main chatting page
+* Make sure the mode selector is in `Local`.  Otherwise if you see `Remote` tap it to switch to `Local`.
+
+For remote inference:
+* Launch a Llama Stack server on localhost (we used fireworks and hf-serverless as the distributor) - [Tutorial](https://github.com/meta-llama/llama-stack-client-kotlin/tree/release/0.0.54?tab=readme-ov-file#setup-remote-inferencing)
+* On another terminal window, `adb reverse tcp:5050 tcp:5050` where 5050 is the port number, change this to whichever one you use/create in your main llama stack flow
+* Select your Model under the Remote Inference section. For example: meta-llama/Llama-3.2-3B-Instruct
+* Provide the Llama Stack server endpoint in Remote URL. For example: http://localhost:5050
+* Go back to the main page 
+* Tap `Local` text button on the top right corner to switch mode to `Remote`
+
+Optional Parameters:
+* Temperature (0.0-2.0): Defaulted to 0, you can adjust the temperature for the model as well. In local mode, the model will automatically reload upon any adjustments. In remote mode, no reload needed.
+* System Prompt: Without any formatting, you can enter in a system prompt. For example, "you are a travel assistant" or "give me a response in a few sentences". Without providing any system prompt will result in the app using default system prompt which is currently set for tool calling capability. 
+
+<p align="center">
+<img src="./docs/img/settings.png" style="width:300px">
+</p>
+
+
+### User Prompt
+Once the model is successfully loaded then enter any prompt and click the send (i.e. generate) button to send it to the model.
+
+You can provide more follow-up questions as well. The current application stores 2 rounds of previous conversations as memory. This can be adjusted with the constant `CONVERSATION_HISTORY_MESSAGE_LOOKBACK`. Since it will append message history as part of the new prompt, this will affect your time-to-first-token. For local inference, the prefill rate is usually smaller than remote inference, which means the model takes longer to process the input prompt. We recommend 2 rounds of conversation history for local inference as a moving window approach.
+
+To understand more about how we are running the inference, check the sample code in `LlamaStackLocalInference.kt` or `LlamaStackRemoteInference.kt`
+
+
+
+### Tool Calling

```

Comment:
indicate this only works for remote at the moment

Reply:
Added in the section